Source/Core/IO/Logging/ConsoleLogger.cs

```diff
@@ -79,61 +73,43 @@ public void Write(LogSeverity severity, string value)
         }
 
         /// <inheritdoc/>
-        public void Write(LogSeverity severity, string format, params object[] args)
+        public override void Write(string value)
         {
-            string value = string.Format(format, args);
-            this.Write(severity, value);
+            this.Write(LogSeverity.Informational, value);
         }
 
         /// <inheritdoc/>
-        public override void WriteLine(string value)
+        public override void Write(string format, params object[] args)
         {
-            this.WriteLine(LogSeverity.Informational, value);
+            string value = string.Format(format, args);
+            this.Write(LogSeverity.Informational, value);
         }
 
         /// <inheritdoc/>
-        public override void WriteLine(string format, params object[] args)
+        public void Write(LogSeverity severity, string format, params object[] args)
         {
             string value = string.Format(format, args);
-            this.WriteLine(LogSeverity.Informational, value);
+            this.Write(severity, value);
         }
 
         /// <inheritdoc/>
         public void WriteLine(LogSeverity severity, string value)
         {
-            switch (severity)
-            {
-                case LogSeverity.Informational:
-                    if (this.LogLevel <= LogSeverity.Informational)
-                    {
-                        Console.WriteLine(value);
-                    }
-
-                    break;
-                case LogSeverity.Warning:
-                    if (this.LogLevel <= LogSeverity.Warning)
-                    {
-                        Error.Write(Console.Out, ConsoleColor.Yellow, value);
-                        Console.WriteLine();
-                    }
-
-                    break;
-                case LogSeverity.Error:
-                    if (this.LogLevel <= LogSeverity.Error)
-                    {
-                        Error.Write(Console.Out, ConsoleColor.Red, value);
-                        Console.WriteLine();
-                    }
-
-                    break;
+            this.Write(severity, value);
+            Console.WriteLine();
```

Comment:
@ShreyTiwari we should avoid having a `Console.WriteLine();` on its own, because now this can print a new line even if the severity does not match!

One option is to add a `\n` in the string passed to `Write` but this will create unnecessary string allocations, and because Logging can be very common and on the hot path, it can hurt performance. Another option is to have an internal method that deals with the `switch` logic and the actual printing and include something like: `if (isNewLine) { Console.WriteLine(); }`, but that will add an `if` condition in every single logging operation which can again be on the hot path. Maybe that's OK but we should do perf analysis, and I feel its low priority for now.

This was the main reason behind the duplication. In general we really want to avoid duplication, with the exception of these kind of perf-critical code pieces where its OK to duplicate code if it will lead to more performance.

You could change the PR to just include the missing `Write` method for consistency, and keep the duplicate code?

Reply:
Yes that is correct, I thought that it would be a very rare case where the default condition of the switch statement is hit and so maybe it is okay to have that new line printed in those scenarios. Anyway, since we do not want to complicate the code path and uphold the performance, I think its best to revert the changes. 

I will update the PR... thanks!

Reply:
Sounds good, thanks @ShreyTiwari!